server/internal/background/activities/report_billing_usage.go

```diff
@@ -0,0 +1,102 @@
+package activities
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"sync"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/speakeasy-api/gram/server/internal/background/activities/repo"
+	"github.com/speakeasy-api/gram/server/internal/billing"
+	"github.com/speakeasy-api/gram/server/internal/mv"
+	"github.com/speakeasy-api/gram/server/internal/oops"
+	orgRepo "github.com/speakeasy-api/gram/server/internal/organizations/repo"
+	"github.com/speakeasy-api/gram/server/internal/thirdparty/posthog"
+)
+
+const (
+	logKeyOrgID     = "org_id"
+	logKeyOrgName   = "org_name"
+	logKeyToolCalls = "tool_calls"
+	logKeyServers   = "servers"
+)
+
+type ReportBillingUsage struct {
+	logger        *slog.Logger
+	billingRepo   billing.Repository
+	repo          *repo.Queries
+	orgRepo       *orgRepo.Queries
+	posthogClient *posthog.Posthog
+}
+
+func NewReportBillingUsage(logger *slog.Logger, db *pgxpool.Pool, billingRepo billing.Repository, posthogClient *posthog.Posthog) *ReportBillingUsage {
+	repo := repo.New(db)
+	orgRepo := orgRepo.New(db)
+
+	return &ReportBillingUsage{
+		logger:        logger,
+		billingRepo:   billingRepo,
+		repo:          repo,
+		orgRepo:       orgRepo,
+		posthogClient: posthogClient,
+	}
+}
+
+// Report billing usage for a list of organizations
+// Send usage data to posthog for tracking purposes
+
+func (r *ReportBillingUsage) Do(ctx context.Context, orgIDs []string) error {
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(orgIDs))
+
+	for _, orgID := range orgIDs {
+		wg.Add(1)
+		go func(orgID string) {
+			defer wg.Done()
+
+			org, err := mv.DescribeOrganization(ctx, r.logger, r.orgRepo, r.billingRepo, orgID)
+			if err != nil {
+				errChan <- fmt.Errorf("failed to describe organization %s: %w", orgID, err)
+				return
+			}
+
+			usage, err := r.billingRepo.GetPeriodUsage(ctx, orgID)
```

Comment:
This is serving a dual purpose right? This is what's populating the cache for the MCP servers to read? MAybe worth calling that out in a comment